scripts/rewrite-declaration-statements-for-typescript.js

```diff
@@ -140,72 +149,76 @@ function transform(srcFile, rootDir) {
-  let resolvedSource = source;
-  const basedir = path.dirname(srcFile);
   // TODO: This section of code will need to be revisited when a third party module is used internally.
-  // currently we haven't built interal dialog or drawer to actually use this rewrite correctly.
-  if (isThirdPartyModule(source)) {
-    // for focus-trap
-    assert(source.indexOf('@material') < 0, '@material/* import sources should have already been rewritten');
-    patchDefaultImportIfNeeded(node);
-    resolvedSource = `${THIRD_PARTY_PATH}${camelCase(source)}`;
-    return resolvedSource;
+  // currently we haven't built internal dialog or drawer to actually use this rewrite correctly.
+  // Needed for focus-trap.
+  if (isThirdPartyModule(oldImportPath)) {
+    if (oldImportPath.indexOf('@material') > -1) {
+      return oldImportPath;
+    }
+    patchDefaultImportIfNeeded(astNode);
+    newImportPath = `${THIRD_PARTY_PATH}${camelCase(oldImportPath)}`;
+    return newImportPath;
   } else {
-    const fileDir = resolve.sync(source, {
+    if (alreadyRewrittenImportPaths.has(oldImportPath)) {
+      return oldImportPath;
+    }
+    const fileDir = resolve.sync(oldImportPath, {
       basedir,
       extensions: ['.ts', '.js'],
     });
-    const srcDirectory = path.dirname(srcFile);
-    resolvedSource = path.relative(srcDirectory, fileDir);
-    const packageStr = resolvedSource
-      .replace('.js', '')
-      .replace('.ts', '');
-    return `./${packageStr}`;
+    const srcDirectory = path.dirname(srcFilePathAbsolute);
+    newImportPath = './' + getBazelFileNameOrPath(
+      path.relative(srcDirectory, fileDir)
+        .replace('.js', '')
+        .replace('.ts', '')
+    );
+    alreadyRewrittenImportPaths.add(newImportPath);
+    return newImportPath;
   }
 }
 
-function patchDefaultImportIfNeeded(node) {
+function patchDefaultImportIfNeeded(astNode) {
   const defaultImportSpecifierIndex =
-      node.specifiers.findIndex(types.isImportDefaultSpecifier);
+      astNode.specifiers.findIndex(types.isImportDefaultSpecifier);
   if (defaultImportSpecifierIndex >= 0) {
-    const defaultImportSpecifier = node.specifiers[defaultImportSpecifierIndex];
+    const defaultImportSpecifier = astNode.specifiers[defaultImportSpecifierIndex];
     const defaultPropImportSpecifier = types.importSpecifier(defaultImportSpecifier.local, types.identifier('default'));
-    node.specifiers[defaultImportSpecifierIndex] = defaultPropImportSpecifier;
+    astNode.specifiers[defaultImportSpecifierIndex] = defaultPropImportSpecifier;
   }
 }
 
+function isThirdPartyModule(importPath) {
+  // See: https://nodejs.org/api/modules.html#modules_all_together (step 3)
+  const wouldLoadAsFileOrDir = ['./', '/', '../'].some((s) => importPath.indexOf(s) === 0);
+  return !wouldLoadAsFileOrDir;
+}
+
+function getBazelFileNameOrPath(ossFileNameOrPath) {
+  return ossFileNameOrPath
+    .replace(/mdc-?/g, '')
+    .replace(/material-components-web/g, 'material_components_web')
```

Comment:
I don't think we even mirror this package?

Reply:
Oh good point. I'll remove it.

Reply:
Removed